.github/workflows/node.js.yml

```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean install of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - run: npm ci
+    - run: node node_modules/puppeteer/install.js
```

Comment:
Why is this needed? Shouldn't puppeteer take care of this itself in its postinstall step?

Reply:
Unfortunately, it seems there's some caching in Github Actions where puppeteer's chromium doesn't get downloaded and it breaks tests.  You can see it in https://github.com/albertyw/mocaccino.js/runs/4604739106?check_suite_focus=true.  From what I'm reading about puppeteer, this script is a postinstall of puppeteer and will have puppeteer download chromium to a directory outside `node_modules` but this makes Github Actions's default caching incomplete.

Reply:
I'm surprised to see the failure only happens on Node 16 (14 and 12 pass). Any idea why that is? Also: the failed test run seems to be from before removing that default cache behavior which makes me wonder if it's related.

Mochify, which obviously also downloads Puppeteer for tests doesn't do anything like this: https://github.com/mantoni/mochify.js/blob/master/.github/workflows/test.yml so it might be nice to see if we can get it working without manually running that script from the hardcoded path.

Reply:
You can actually skip installing chromium entirely since the GitHub action container already comes with chrome installed.

```
      - name: Install
        run: npm ci
        env:
          PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: 1
      - name: Test
        run: npm test -- --chrome $(which google-chrome-stable) --allow-chrome-as-root
```

Reply:
Using this configuration (https://github.com/albertyw/mocaccino.js/commit/412e515f19089c87cbc4aad25796d76525d767af) doesn't work either, with
```
     Error: Could not find expected browser (chrome) locally. Run `npm install` to download the correct Chromium revision (869685).
      at ChromeLauncher.launch (node_modules/puppeteer/lib/cjs/puppeteer/node/Launcher.js:82:27)
```

on all three node versions.

Reply:
Did you try:

- no caching at all
- no manual calling of the preinstall script
- no extra flags

This is what Mochify is doing so I would be surprised if it doesn't work here. See: https://github.com/mantoni/mochify.js/pull/225

Reply:
If you're referring to
```
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v2
      with:
        node-version: ${{ matrix.node-version }}
    - run: npm ci
    - run: npm test
```
which is equivalent to mantoni/mochify.js#225, that results in the original issue which is Node 12 and 14 pass but Node 16 have failures to find chromium: https://github.com/albertyw/mocaccino.js/runs/4621105656

Reply:
I can actually repro this locally on npm 7 / Node 14. Like everything npm always, it works again after nuking node_modules and the lockfile:

```
➜  mocaccino.js git:(github-actions) npm -v
7.24.2
➜  mocaccino.js git:(github-actions) node -v
v14.15.5
➜  mocaccino.js git:(github-actions) npm ci

added 389 packages, and audited 390 packages in 3s

12 vulnerabilities (10 moderate, 2 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
➜  mocaccino.js git:(github-actions) ls node_modules/puppeteer/.local-chromium
ls: Zugriff auf 'node_modules/puppeteer/.local-chromium' nicht möglich: Datei oder Verzeichnis nicht gefunden
➜  mocaccino.js git:(github-actions) rm package-lock.json 
➜  mocaccino.js git:(github-actions) ✗ rm -rf node_modules 
➜  mocaccino.js git:(github-actions) ✗ npm i 
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated mkdirp@0.5.0: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)

added 400 packages, and audited 401 packages in 30s

29 packages are looking for funding
  run `npm fund` for details

10 vulnerabilities (8 moderate, 2 high)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
➜  mocaccino.js git:(github-actions) ✗ ls node_modules/puppeteer/.local-chromium
linux-869685
```

Reply:
I haven't been able to reproduce locally.  Is there some other Github Actions configuration to try or should we go back to the original (albeit inefficient) `node node_modules/puppeteer/install.js`?

Reply:
I think we should still try recreating that lockfile from scratch and see if this helps in CI as well.

Reply:
Looks like that worked: https://github.com/albertyw/mocaccino.js/runs/4625134174